app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/StoreRecordsController.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesdatastore.controllers
+
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.tradergoodsprofilesdatastore.connectors.RouterConnector
+import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.IdentifierAction
+import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.{CheckRecordsRepository, RecordsRepository}
+import org.apache.pekko.Done
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import play.api.mvc.Request
+
+class StoreRecordsController @Inject() (
+  recordsRepository: RecordsRepository,
+  checkRecordsRepository: CheckRecordsRepository,
+  cc: ControllerComponents,
+  routerConnector: RouterConnector,
+  identify: IdentifierAction
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc) {
+
+  def storeLatestRecords(
+    eori: String
+  ): Action[AnyContent] = identify.async { implicit request =>
+    recordsRepository.getLatest(eori).flatMap {
+      case Some(record) =>
+        storeRecordsRecursively(eori, 1, Some(record.updatedDateTime.plusSeconds(1).toString), 0)
```

Comment:
What's the plus one second up to?

Reply:
because the time is inclusive so if i send back the exact time i get that record again which i don't want i want all the records that were updated after it - can you think of a better way or do i just duplicate that record? well it's not a duplicate because it patches it - but maybe that's better?

Reply:
Gotcha ok. Might be better. Now I'm thinking about if the next record is like 999 milliseconds after the last one and we skip it because we go a second. 
If the patch is side-effect free then I think that's nicer.

Reply:
removed